### DISCUSSION TITLE
Query API vector output when using DiskANN

### DISCUSSION THREAD

**Discussion by scurkovic** [milvus-io/milvus#26947] [Q&A and General discussion] [answered]
Hi,

Is it possible to use the Query API to return the raw vectors with a DiskANN? [This](https://github.com/milvus-io/milvus/discussions/24787#discussioncomment-6158362) comment suggests that it is, but when I try it I get an error:

> RPC error: [query], <MilvusException: (code=1, message=attempt #0: fail to Query, QueryNode ID = 1, reason=failed to query channel, err=worker(1) query failed: vector output fields for DISKANN index is not allowed: channel=by-dev-rootcoord-dml_1_444116693516689307v0: fail to search/query shard leader: attempt #1: no available shard delegator found: service unavailable: fail to query on all shard leaders)>, <Time:{'RPC start': '2023-09-08 14:03:06.600723', 'RPC error': '2023-09-08 14:03:07.228388'}>

I found no issues or discussions referencing this. I'm using the latest `docker-compose.yml`. The same thing happens with the Search API. It's also worth mentioning that the index is still building.

**Comment by xiaofan-luan** [comment]
> Hi,
> 
> Is it possible to use the Query API to return the raw vectors with a DiskANN? [This](https://github.com/milvus-io/milvus/discussions/24787#discussioncomment-6158362) comment suggests that it is, but when I try it I get an error:
> 
> > RPC error: [query], <MilvusException: (code=1, message=attempt #0: fail to Query, QueryNode ID = 1, reason=failed to query channel, err=worker(1) query failed: vector output fields for DISKANN index is not allowed: channel=by-dev-rootcoord-dml_1_444116693516689307v0: fail to search/query shard leader: attempt #1: no available shard delegator found: service unavailable: fail to query on all shard leaders)>, <Time:{'RPC start': '2023-09-08 14:03:06.600723', 'RPC error': '2023-09-08 14:03:07.228388'}>
> 
> I found no issues or discussions referencing this. I'm using the latest `docker-compose.yml`. The same thing happens with the Search API. It's also worth mentioning that the index is still building.

I think this is already supported on latest 2.2.14

**Reply by scurkovic** [reply]
I am running version 2.3.0

**Comment by xiaofan-luan** [comment, marked as answer]
if using DiskANN IP metrics，current implementation is not supported to retrieve vector field.

1. we are gonna to support it later this month
2. the error messaage need to be refined.

@liliu-z 
@jiaoew1991

**Comment by liliu-z** [comment]
Made a issue to track: https://github.com/zilliztech/knowhere/issues/67
